File: c/arvore_avl/menuAvl.c
#include <stdio.h>
#include "menuAvl.h"
#include "../arvore_avl/avlTree.h"
#include "../common/sobreOpcoes.h"

void menuAvl() {
	AvlTree * avl = newAvl();
	int opcao, item;
	do {
		printf("\n");
		printf("\t\tMENU: ARVORE AVL\n");
		printf("  [1] - inserir item na árvore\n");
		printf("  [2] - remover item da árvore\n");
		printf("  [3] - checar se existe item na árvore\n");
		printf("  [4] - pesquisar informações do nó que contém determinado item\n");
		printf("  [5] - mostrar árvore em todas as ordens\n");
		printf("  [6] - mostrar árvore na pré-ordem\n");
		printf("  [7] - mostrar árvore em-ordem\n");
		printf("  [8] - mostrar árvore na pós-ordem\n");
		printf("  [9] - mostrar número de nós da árvore\n");
		printf("  [0] - sair\n");
		printf("  Opção: ");
		opcao = lerOpcao();
		opcao = tratarOpcao(opcao, 0, 9);
		printf("\n");
		switch(opcao) {
			case 0:
				desalocarArvoreDaMemoria(avl);
				printf("Programa encerrado.\n");
				break;
			case 1:
				printf("Item a ser inserido: ");
				scanf("%d", &item);
				if (inserirNaArvore(avl, item)) {
					printf("Item inserido com sucesso.\n");
				} else { 
					printf("Falha ao inserir o item.\n");
				}
				break;
			case 2:
				printf("Item a ser removido: ");
				scanf("%d", &item);
				if (removerDaArvore(avl, item)) {
					printf("Item removido com sucesso.\n");
				} else { 
					printf("Falha ao remover o item.\n");
				}
				break;
			case 3:
				printf("Item a ser procurado: ");
				scanf("%d", &item);
				if (pesquisarPorItemNaArvore(avl, item)) {
					printf("Item encontrado na árvore.\n");
				} else {
					printf("Item não encontrado.\n");
				}
				break;
			case 4:
				printf("Item a ser procurado: ");
				scanf("%d", &item);
				pesquisarPorNoContendoItem(avl, item);
				break;
			case 5:
				mostrarTodasOrdens(avl);
				break;
			case 6:
				mostrarPreOrdem(avl);
				break;
			case 7:
				mostrarEmOrdem(avl);
				break;
			case 8:
				mostrarPosOrdem(avl);
				break;
			case 9:
				printf("Número de nós: %d\n", contarNosDaArvore(avl));
				break;
			default:
				break;
		}
	} while (opcao != 0);
}

File: java/arvore_avl/MenuAvl.java
package arvore_avl;

import main.Common;
import main.MenuInterface;

public class MenuAvl implements MenuInterface {
	@Override
	public void menu() {
		ArvoreAvl avl = new ArvoreAvl();
		int opcao, item;
		do {
			System.out.println("");
			System.out.println("\t\tMENU: ARVORE AVL");
			System.out.println("  [1] - inserir item na árvore");
			System.out.println("  [2] - remover item da árvore");
			System.out.println("  [3] - checar se existe item na árvore");
			System.out.println("  [4] - pesquisar informações do nó que contém determinado item");
			System.out.println("  [5] - mostrar árvore em todas as ordens");
			System.out.println("  [6] - mostrar árvore na pré-ordem");
			System.out.println("  [7] - mostrar árvore em-ordem");
			System.out.println("  [8] - mostrar árvore na pós-ordem");
			System.out.println("  [9] - mostrar número de nós da árvore");
			System.out.println("  [0] - sair");
			System.out.print("  Opção: ");
			opcao = Common.input.nextInt();
			System.out.println("");
			switch(opcao) {
				case 0:
					avl.desalocarArvoreDaMemoria();
					System.out.println("Programa encerrado.");
					break;
				case 1:
					System.out.print("Item a ser inserido: ");
					item = Common.input.nextInt();
					if (avl.inserirItemNaAvl(item)) {
						System.out.println("Item inserido com sucesso.");
					} else { 
						System.out.println("Falha ao inserir o item.");
					}
					break;
				case 2:
					System.out.print("Item a ser removido: ");
					item = Common.input.nextInt();
					if (avl.removerItemDaAvl(item)) {
						System.out.println("Item removido com sucesso.");
					} else { 
						System.out.println("Falha ao remover o item.");
					}
					break;
				case 3:
					System.out.print("Item a ser procurado: ");
					item = Common.input.nextInt();
					if (avl.pesquisarPorItemNaAvl(item)) {
						System.out.println("Item encontrado na árvore.");
					} else {
						System.out.println("Item não encontrado.");
					}
					break;
				case 4:
					System.out.print("Item a ser procurado: ");
					item = Common.input.nextInt();
					avl.obterInformacoesDoNo(item);
					break;
				case 5:
					avl.mostrarEmTodasAsOrdens();
					break;
				case 6:
					avl.mostrarAvlEmPreOrdem();
					break;
				case 7:
					avl.mostrarAvlEmOrdem();
					break;
				case 8:
					avl.mostrarAvlEmPosOrdem();
					break;
				case 9:
					System.out.println("Número de nós: "+avl.contarNosDaArvore());
					break;
				default:
					break;
			}
		} while (opcao != 0);
	}
}
